tomcat9x/src/main/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServer.groovy

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.bmuschko.gradle.tomcat.embedded
+
+import java.lang.reflect.Constructor
+
+/**
+ * Tomcat 9x server implementation.
+ *
+ * @author Benjamin Muschko
+ * @author Andrey Bloschetsov
+ * @author Martin Flower
+ * @author Keisuke Kato
+ */
+class Tomcat9xServer extends BaseTomcat7xPlusImpl {
```

Comment:
It looks to me as if `Tomcat8xServer`, `Tomcat85xServer` and `Tomcat9xServer` use pretty much the same code. Let's unify these classes. I believe the only difference is the return value of `getVersion()`. Could you break out another parent class? In the long we'll probably want to rather go for a delegation pattern.

Reply:
@bmuschko I created BaseTomcat8xPlusImpl.